### PR TITLE
Docs: Fix broken link in the quickstart and fix error in code example, capitalize S in TypeScript

### PR DESCRIPTION
**Pull request by camiinthisthang**
- update a broke link
- update the test to reflect changes made in the Rust SDK
- fix return statement in example code

### CODE REVIEW

File: src/developer-quickstart.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ A script is runnable bytecode on the chain which can call contracts to perform s
 | script    | ❌                                   | ❌                    | ❌                           | ❌                        |
 | library   | ✅ (via a contract or predicate) | ❌ | ❌                           | ✅                       |
 
-See [the chapter on program types](../sway-program-types/index.md) for more information.
+See [the chapter on program types](https://fuellabs.github.io/sway/v0.19.2/sway-program-types/index.html) for more information.
```
Comment (mohammadfawaz) [inline src/developer-quickstart.md]: Should we use the latest version here? We sadly can't use `latest` in the link above but we can use `mater`:
```suggestion
See [the chapter on program types](https://fuellabs.github.io/sway/master/sway-program-types/index.html) for more information.
```

File: src/developer-quickstart.md [hunk 2]
```diff
@@ -186,7 +186,7 @@ abi Counter {
 impl Counter for Contract {
     #[storage(read)]
     fn count() -> u64 {
-      return storage.counter;
+      storage.counter
     }
```
Comment (mohammadfawaz) [inline src/developer-quickstart.md:190]: Maybe a line break between the two methods:
```suggestion
    }

```

File: src/developer-quickstart.md [hunk 3]
```diff
@@ -186,7 +186,7 @@ abi Counter {
 impl Counter for Contract {
     #[storage(read)]
     fn count() -> u64 {
-      return storage.counter;
+      storage.counter
```
Comment (mohammadfawaz) [inline src/developer-quickstart.md]: Just 2 more spaces to get the right indentation :)
```suggestion
        storage.counter
```

File: src/developer-quickstart.md [hunk 4]
```diff
@@ -186,7 +186,7 @@ abi Counter {
...
     }
     #[storage(read, write)]
     fn increment(){
```
Comment (mohammadfawaz) [inline src/developer-quickstart.md]: Extra space after `()`
```suggestion
    fn increment() {
```

**Review by mohammadfawaz — COMMENTED** [review]
Thanks @camiinthisthang! 

Recently, I made changes to the testing section here: https://github.com/FuelLabs/sway/pull/2737/files - do you mind replicating them here as well?

**Comment by camiinthisthang** [comment]
The quickstart here deploys to the testnet, and AFAIK transactions will revert since they are not signed on testnet. @mohammadfawaz, do I have this right?

**Comment by mohammadfawaz** [comment]
> The quickstart here deploys to the testnet, and AFAIK transactions will revert since they are not signed on testnet. @mohammadfawaz, do I have this right?

You're right we should not be adding `--unsigned` here! I was actually specifically talking about the changes to the Rust SDK test under "Testing your Contract". I.e. replacing
```rust
    let (instance, _id) = get_contract_instance().await;
    // Now you have an instance of your contract you can use to test each function
    let result = instance.increment().call().await.unwrap();
    assert!(result.value > 0)
```
with
```rust
    // Increment the counter
    let _result = instance.increment().call().await.unwrap();

    // Get the current value of the counter
    let result = instance.counter().call().await.unwrap();
    assert!(result.value > 0);
```

**Comment by camiinthisthang** [comment]
oh perf. Done, thank you @mohammadfawaz !

File: src/developer-quickstart.md [hunk 5]
```diff
@@ -216,10 +217,10 @@ At the bottom of the file, define the body of `can_get_contract_instance`. Here
...
 #[tokio::test]
 async fn can_get_contract_id() {
-    let (instance, _id) = get_contract_instance().await;
-    // Now you have an instance of your contract you can use to test each function
-    instance.increment().call().await.unwrap();
-    let result = instance.count().call().await.unwrap();
+    // Increment the counter
+    let _result = instance.increment().call().await.unwrap();
+    // Get the current value of the counter
+    let result = instance.counter().call().await.unwrap();
```
Comment (mohammadfawaz) [inline src/developer-quickstart.md]:
```suggestion
    let result = instance.count().call().await.unwrap();
```

Comment (mohammadfawaz) [reply src/developer-quickstart.md]: This one is totally on me - sorry.

**Review by mohammadfawaz — DISMISSED** [review]
:shipit: Great! Thank you :)

File: src/developer-quickstart.md [hunk 6]
```diff
@@ -216,9 +217,9 @@ At the bottom of the file, define the body of `can_get_contract_instance`. Here
 ```sway
 #[tokio::test]
 async fn can_get_contract_id() {
-    let (instance, _id) = get_contract_instance().await;
-    // Now you have an instance of your contract you can use to test each function
-    instance.increment().call().await.unwrap();
+    // Increment the counter
```
Comment (mohammadfawaz) [inline src/developer-quickstart.md:221]:
```suggestion
    let (instance, _id) = get_contract_instance().await;
    // Increment the counter
```

**Review by mohammadfawaz — APPROVED** [review]

**Pull request merged**